### PR TITLE
Add SpdmSecurityLib for SecurityPkg 

### PR DESCRIPTION
**Pull request by qizhangz**

### CODE REVIEW

File: SecurityPkg/Include/Library/SpdmSecurityLib.h [hunk 1]
```diff
@@ -273,115 +273,115 @@ typedef SPDM_RETURN (*SPDM_DEVICE_ACQUIRE_RECEIVER_BUFFER_FUNC)(
...
+*/
+RETURN_STATUS
+EFIAPI
+SpdmDeviceAuthenticationAndMeasurement (
+  IN  EDKII_SPDM_DEVICE_INFO        *SpdmDeviceInfo,
+  IN  EDKII_DEVICE_SECURITY_POLICY  *SecurityPolicy,
+  OUT EDKII_DEVICE_SECURITY_STATE   *SecurityState
+  );
+
+VOID *
```
Comment (jyao1) [inline SecurityPkg/Include/Library/SpdmSecurityLib.h]: miss EFIAPI

File: SecurityPkg/Include/Library/SpdmSecurityLib.h [hunk 2]
```diff
@@ -117,11 +117,11 @@ typedef SPDM_RETURN (*SPDM_DEVICE_RECEIVE_MESSAGE_FUNC)(
  * @retval RETURN_INVALID_PARAMETER     The message is NULL or the message_size is zero.
  **/
 typedef SPDM_RETURN (*SPDM_TRANSPORT_ENCODE_MESSAGE_FUNC)(
```
Comment (jyao1) [inline SecurityPkg/Include/Library/SpdmSecurityLib.h]: Coding style:
typedef
SPDM_RETURN
EFIAPI
(*SPDM_TRANSPORT_ENCODE_MESSAGE_FUNC)

File: SecurityPkg/Include/Library/SpdmSecurityLib.h [hunk 3]
```diff
@@ -154,10 +154,10 @@ typedef SPDM_RETURN (*SPDM_TRANSPORT_ENCODE_MESSAGE_FUNC)(
  **/
 typedef SPDM_RETURN (*SPDM_TRANSPORT_DECODE_MESSAGE_FUNC)(
     VOID *SpdmContext, UINT32 **SessionId,
-    BOOLEAN *IsAppMessage, BOOLEAN IsRequester,
+    bool *IsAppMessage, bool IsRequester,
```
Comment (jyao1) [inline SecurityPkg/Include/Library/SpdmSecurityLib.h]: need BOOLEAN

File: SecurityPkg/Include/Library/SpdmSecurityLib.h [hunk 4]
```diff
@@ -222,7 +222,7 @@ typedef SPDM_RETURN (*SPDM_DEVICE_ACQUIRE_SENDER_BUFFER_FUNC)(
  **/
 typedef void (*SPDM_DEVICE_RELEASE_SENDER_BUFFER_FUNC)(
```
Comment (jyao1) [inline SecurityPkg/Include/Library/SpdmSecurityLib.h]: need VOID

File: SecurityPkg/SecurityPkg.dec [hunk 5]
```diff
@@ -21,6 +21,7 @@
 
 [Includes]
   Include
+  ../DeviceSecurityPkg/Library/SpdmLib/Include
```
Comment (jyao1) [inline SecurityPkg/SecurityPkg.dec]: how can that work?

**Comment by qizhangz** [comment]
has passed All tianocore CI.
https://github.com/tianocore/edk2/pull/3725

**Review by jyao1 — APPROVED** [review]

**Pull request merged**